tests/unit/infrastructure/database/repositories/test_sqlalchemy_speaker_embedding_repository.py

```
@@ -0,0 +1,319 @@
+
+    async def test_add_rollback_on_error(
+        self,
+        async_repo: AsyncSQLAlchemySpeakerEmbeddingRepository,
+        async_session: AsyncMock,
+    ) -> None:
+        """Test add rolls back on SQLAlchemy error."""
+        async_session.commit.side_effect = SQLAlchemyError("DB error")
+        embedding = SpeakerEmbedding(
+            uuid="test-uuid", speaker_label="Alice", embedding=[0.1]
+        )
+        with pytest.raises(Exception):
```

Ruff (0.15.14)

[warning] Do not assert blind exception: `Exception`

(B017)